src/components/select/simple-select/simple-select.test.tsx

```diff
@@ -1,85 +1,794 @@
+    <SimpleSelect label="Colour" onChange={() => {}} onFocus={onFocus}>
+      <Option text="amber" value="amber" />
+    </SimpleSelect>
+  );
+
+  await user.tab();
+
+  expect(onFocus).toHaveBeenCalledTimes(1);
+});
+
+describe("onBlur prop", () => {
+  it("calls onBlur callback when input loses focus", async () => {
+    const onBlur = jest.fn();
+    const user = userEvent.setup();
+    render(
+      <SimpleSelect label="Colour" onChange={() => {}} onBlur={onBlur}>
+        <Option text="amber" value="amber" />
+      </SimpleSelect>
+    );
+
+    await user.tab();
+    await user.tab({ shift: true });
+
+    expect(onBlur).toHaveBeenCalledTimes(1);
+  });
+
+  it("does not call onBlur when input loses focus temporarily due to option selection", async () => {
+    const onBlur = jest.fn();
+    const user = userEvent.setup();
+    render(
+      <SimpleSelect label="Colour" onChange={() => {}} onBlur={onBlur}>
+        <Option text="amber" value="amber" />
+      </SimpleSelect>
+    );
+
+    await user.click(screen.getByRole("combobox"));
+
+    await user.click(screen.getByRole("option", { name: "amber" }));
+
+    expect(onBlur).not.toHaveBeenCalled();
+  });
+});
+
+describe("forwarded ref", () => {
+  it("allows access to the input element through a forwarded callback ref", () => {
+    const mockRef = jest.fn((element) => element);
+    render(
+      <SimpleSelect label="Colour" onChange={() => {}} ref={mockRef}>
+        <Option text="amber" value="amber" />
+      </SimpleSelect>
+    );
+
+    expect(mockRef).toHaveBeenNthCalledWith(1, screen.getByRole("combobox"));
+  });
+
+  it("allows access to the input element through a forwarded ref object", () => {
+    const mockRef = { current: null };
+    render(
+      <SimpleSelect label="Colour" onChange={() => {}} ref={mockRef}>
+        <Option text="amber" value="amber" />
+      </SimpleSelect>
+    );
+
+    expect(mockRef.current).toBe(screen.getByRole("combobox"));
+  });
+});
+
+test("raises deprecation warning when component is used with defaultValue and no onChange (uncontrolled usage)", () => {
+  jest.spyOn(console, "warn").mockImplementation(() => {});
+
+  const loggerSpy = jest.spyOn(Logger, "deprecate");
+  render(
+    <SimpleSelect label="Colour" onChange={undefined} defaultValue="amber">
+      <Option text="amber" value="amber" />
+    </SimpleSelect>
+  );
+
+  expect(loggerSpy).toHaveBeenNthCalledWith(
+    1,
+    "Uncontrolled behaviour in `Simple Select` is deprecated and support will soon be removed. Please make sure all your inputs are controlled."
+  );
+});
+
+describe("required prop", () => {
+  it("marks input as required when required prop is true", () => {
+    render(
+      <SimpleSelect label="Colour" onChange={() => {}} required>
+        <Option text="amber" value="amber" />
+      </SimpleSelect>
+    );
+
+    expect(screen.getByRole("combobox")).toBeRequired();
+  });
+
+  it("adds asterisk to label when required prop is true", () => {
```

Comment:
comment (non-blocking): we could remove this and the one below as they are captured by chromatic